### ISSUE TITLE
Add a setting to remove the "Administrator:" prefix from tab titles

### ISSUE THREAD

**Issue by DJackman123** [microsoft/terminal#984]
# Summary of the new feature/enhancement

When I run Terminal as administrator, every tab has "Administrator:" added as a prefix to the (already long) title.  I know it's running as administrator and (at least for now) all the tabs are admin sessions.  I want the tab titles to be as short as possible so I can see more at once.  Being able to remove this administrator prefix will help me do that.

# Proposed technical implementation details (optional)

**Comment by zadjii-msft** [comment]
Yea, there's gonna be like no way to do this without setting custom title overrides (#608). The "Administrator: " text comes from both cmd.exe and powershell.exe, not from the Terminal itself. Note that it goes away if you open a WSL shell, because linux shells don't display "Administrator" in the title.

As far as I'm aware, you can't tell cmd.exe to _not_ display that text when running elevated.

It's entirely possible that there's a way to tell powershell to not display that text, though I don't know enough about powershell to know for sure. Summoning @adiviness and @DHowett-MSFT

**Comment by adiviness** [comment]
powershell lets you have a custom prompt, I presume that you could change the title from within that function.

**Comment by binarycrusader** [comment]
cmd has the "title" command but even that doesn't let you override the Administrator text.  Personally, I understand the security implications and why it might be undesirable.

**Comment by oising** [comment]
@djackman123 A quick way to remove the prefix for powershell is to put the following into your profile.ps1 file (learn more with `get-help about_profiles | more`):

`[console]::title = [console]::title.replace("Administrator:", "")`

For cmd shells, it's a little more involved, and when I say a "little," I mean downright unreasonably difficult to do. There is no direct way to alter the console title from batch, and only by emitting VT sequences can you change it - even then, you can only effect a permanent change to it *interactively*. When you run a batch file, or another process, the current window title is pushed (saved) onto a stack and then changed to reflect the running process/command, and then popped (restored) when your command/batch is done. And example -- run the following commands in powershell to create a batch file that changes the window title:

```powershell
echo "@echo.`e]2;new title`a" > init.cmd
echo "@pause" >> init.cmd 
```
Now open a cmd shell, and run `init.cmd`. You'll see the title change, but it will restored once you press a key. This is already a blocker and we haven't even tried to read the current title yet to remove the prefix (like we do in powershell.)

Incidentally, if you wanted to emulate the above script using pure batch without a dependency on powershell's handy syntax for non-printable characters, it looks like this - hold onto your lunch:

**settitle.cmd**
```cmd
@echo off & setlocal & setlocal enabledelayedexpansion
if "%1" == "" goto error
set TABNAME=%*
for /f "tokens=* delims=" %%a in ('forfiles /p %~dps0 /m %~nxs0 /c "cmd /c echo.0x1B0x07"') do @set arrChar=%%a
set ESC=!arrChar:~0,1!
set BEL=!arrChar:~1,1!
echo.%ESC%]0;%TABNAME%%BEL%
pause
goto end
:error
echo usage: title tabname
:end
```
Why so horrifying? Because the only way to get ESC and BEL natively in batch is to translate the hex ascii equivalents using forfiles.exe, and then capture into a variable. 

TL;DR -- easy to do in powershell. Hard or impossible to do in cmd. This needs support in the terminal, conhost and the VT engine.

**Comment by DHowett-MSFT** [comment]
> There is no direct way to alter the console title from batch
> TL;DR -- easy to do in powershell. Hard or impossible to do in cmd.

Well, admittedly, there is the `title` command ... but it doesn't stomp the `Administrator:` because it's a CMD builtin and CMD thinks it knows better than you.

![image](https://user-images.githubusercontent.com/14316954/58347059-afc55600-7e11-11e9-8f7e-cf350aaa7727.png)

**Comment by mdtauk** [comment]
When/If a statusbar is added to Windows Terminal - the "Administrator" text could be displayed there instead of taking up valuable space in the Tab itself.  I don't think it is a good idea to omit any indicator to identify the window as being elevated - for security peace of mind.

For Ubuntu, the statusbar could display the current user or **sudo** when the prompt is of higher privilege.

**Comment by DHowett-MSFT** [comment]
@mdtauk I'm not saying it isn't _doable_, but as a consumer of only standard virtual terminal sequences there is no non-heuristic way for Terminal to detect _anything about the connected client_. Administrator context on Windows is a little easier because we can guess based on how we started it, but anything else is a non-starter.

**Comment by DHowett-MSFT** [comment]
This is my primary argument against a status bar, honestly. There's nothing from the client application you can put in it that isn't guessed off a heuristic (user, computer) or simply unknowable (codepage?) to a terminal emulator.

**Comment by mdtauk** [comment]
Sounds like an API that should be there. Something like:

Microsoft.UI.ApplicationModel.IsCurrentlyElevated

**Comment by DHowett-MSFT** [comment]
Yes, that would solve exactly one half of one of the problems I listed. It is, unfortunately, the half that’s already solvable by some other existing means.

**Comment by mdtauk** [comment]
> Yes, that would solve exactly one half of one of the problems I listed. It is, unfortunately, the half that’s already solvable by some other existing means.

I took my inspiration for the statusbar contents from the teaser.  So I assumed these were on the ToDo list.

![image](https://user-images.githubusercontent.com/7389110/58348897-c7c0c800-7e59-11e9-8190-ef5ae92ae1ce.png)

**Comment by oising** [comment]
@DHowett-MSFT Lordy, I forgot about TITLE. I remember using it in 4DOS and 4NT back in the day. Anyway, I guess it doesn't do the trick either as you say.

**Comment by DHowett-MSFT** [comment]
This will be better served by #608.

**Comment by ElektroStudios** [comment]
> As far as I'm aware, you can't tell cmd.exe to _not_ display that text when running elevated.

There is an approach modifying the cmd.exe.mui file:

https://serverfault.com/questions/35561/how-to-remove-administrator-from-the-command-promt-title/1047538#1047538

But the powershell.exe.mui comes more difficult to hack (for my poor knowledge)

**Comment by trallnag** [comment]
For PowerShell I came across this command:

    $host.ui.RawUI.WindowTitle = “Changed Title”

Source: <https://devblogs.microsoft.com/scripting/powertip-change-the-powershell-console-title/>

So if the prefix is really bugging someone you can use this snippet to strip out the prefix.

**Comment by dmitrii-fediuk** [comment]
[A working solution for `cmd`](https://df.tips/t/topic/1566)
